pkg/k8s/pipelineRun.go

```diff
@@ -241,7 +241,7 @@ func (r *pipelineRun) DeleteFinalizerIfExists() error {
 
 func (r *pipelineRun) updateFinalizers(finalizerList []string) error {
 	if r.client == nil {
-		return fmt.Errorf("No factory provided to store updates [%s]", r.String())
```

Comment:
Please update test 'Test_pipelineRun_UpdateResult_noFactory' accordingly.

Reply:
Done. Also added some more test for other functions.